src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4398,13 +4552,17 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
                             }
                             break;
 
+                        case ConversionKind.ImplicitNullable:
+                            nullableAnnotation = (_resultType.IsNullableType() && _resultType.NullableAnnotation.IsAnyNullable()) ? NullableAnnotation.Nullable : NullableAnnotation.NotNullable;
```

Comment:
The helper method would need two parameters since there are several instances where the type and nullability are from distinct TypeSymbolWithAnnotations. Will leave as is for now.

---
In reply to: [239603046](https://github.com/dotnet/roslyn/pull/31499#discussion_r239603046) [](ancestors = 239603046)